### ISSUE TITLE
Social Web Sign In - Return UserAlreadyExists instead of CodeValidationException

### ISSUE THREAD

**Issue by tylerjroach** [aws-amplify/amplify-android#2302]
### Discussed in https://github.com/aws-amplify/amplify-android/discussions/2301

<div type='discussions-op-text'>

<sup>Originally posted by **MrBenjoo** February 22, 2023</sup>
How can one determine if an account already exists when the user is trying to sign in with social web UI but their account already exists?
The account was likely created using Amplify.Auth.confirmSignUp, which means the user will need to sign in with email and password within the app instead. 

Before, using version 1.37.10, I was able to catch this with an ugly check: `PreSignUp failed with error User already exists`. But using the latest version it returns `CodeValidationException{message=invalid_request, cause=null, recoverySuggestion=Sorry, we don't have a suggested fix for this error yet.}`

Would it be possible to return `AliasExistsException` instead, or is there some other way i can catch a suitable exception?

</div>

**Comment by tylerjroach** [comment]
PR has been merged and should make it in the next Amplify release. You will be able to match the error by checking for:

```
CodeValidationException(
  message = "invalid_request: User already exists with provider user id."
)
```

**Comment by MrBenjoo** [comment]
Thanks! Any reason why we have to check the message to determine the issue? My initial tought was to catch something like AccountAlreadyExists or similar.

**Comment by tylerjroach** [comment]
Hi @MrBenjoo The hosted ui responds with an "error" string and an "error description". See the possible list of errors under "Examples of negative responses" in this link https://docs.aws.amazon.com/cognito/latest/developerguide/token-endpoint.html.

We did not provide a specific exception because it would have required matching on both "error" and "error_description" and creating exception types for a wide variety of errors that could be returned. All of these errors result in a `CodeValidationException` for us on the client side.

**Comment by eeatonaws** [comment]
A fix has been released in Amplify Android 2.4.1.
